### PR TITLE
Remove compound `Length` from `alfa-style`

### PR DESCRIPTION
**Pull request by Jym77**
Part of #1202

### CODE REVIEW

**Comment by Jym77** [comment]
!pr extract

File: packages/alfa-css/src/value/numeric/length.ts [hunk 1]
```diff
@@ -117,35 +121,57 @@ export class Length<
...
+  export type Fixed<U extends Unit.Length = Unit.Length> = Length<U, false>;
+  /**
+   * Convenience type for Length that may or not contain calculation
+   */
+  export type Mixed<U extends Unit.Length = Unit.Length> =
+    | Calculated
+    | Fixed<U>;
+
+  export namespace Fixed {
+    export interface JSON<U extends Unit.Length = Unit.Length>
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/numeric/length.ts:164]: **question:** The type parameter doesn't seem to be used. Can it be removed?

File: packages/alfa-css/src/value/numeric/length.ts [hunk 2]
```diff
@@ -117,35 +121,57 @@ export class Length<
...
-  public toJSON(): Length.JSON<CALC, U> {
-    return {
-      ...super.toJSON(),
-      value: this._value ?? null,
-      unit: this._unit ?? null,
-      math: this._math?.toJSON() ?? null,
-    };
+  public toJSON(
+    this: Length.Mixed<U>
+  ): Length.Fixed.JSON<U> | Length.Calculated.JSON {
```
Comment (rcj-siteimprove) [inline packages/alfa-css/src/value/numeric/length.ts:126]: **question:** Same as other comment, can `Length.Fixed.JSON` be made non-generic?

File: packages/alfa-style/src/element/predicate/has-outline.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { Color, Length } from "@siteimprove/alfa-css";
+import { Color, Unit } from "@siteimprove/alfa-css";
```
[the remark this answers is not in the document]
Comment (rcj-siteimprove) [reply packages/alfa-style/src/element/predicate/has-outline.ts:1]: **suggestion:**
```suggestion
import { Color } from "@siteimprove/alfa-css";
```

**Review by rcj-siteimprove — APPROVED** [review]
Looks good, just found some potential polishing

**Pull request merged**